polynote-kernel/src/main/scala/polynote/kernel/util/CellContext.scala

```diff
@@ -0,0 +1,79 @@
+package polynote.kernel.util
+
+import cats.effect.{Concurrent, IO}
+import cats.effect.concurrent.{Deferred, TryableDeferred}
+import polynote.kernel.ResultValue
+import polynote.messages.CellID
+
+import scala.annotation.tailrec
+
+
+class CellContext(
+  val id: CellID,
+  val module: TryableDeferred[IO, scala.reflect.api.Symbols#ModuleSymbol],
```

Comment:
Hmm, I wasn't able to find where this was being used? Is the plan for this to eventually take over `ScalaInterpreter.previousSources`?

Reply:
It's used now to check whether the interpreter defined its own module; if not, then `PolyKernel` creates one to wrap the results.

And yes, I hope to get rid of `previousSources` eventually.